### ISSUE TITLE
Fix FP S6333: 'authorizationType' defined using 'authorizer'

### ISSUE THREAD

**Issue by pedro-oliveira-sonarsource** [SonarSource/SonarJS#3648]
### **What**

In  #3452 , support for _aws_cdk.aws_apigateway.RestApi_ was added, where we raise an issue if an endpoint is defined to be public.

On a validation sprint, when analyzing the [implementation ticket](https://github.com/SonarSource/SonarJS/issues/3452) produced during the research for the rule, a false-positive was detected if the _authorizationType_ property is defined using the _[authorizer](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-apigateway.MethodOptions.html#authorizer)_ property available in the _[MethodOptions](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-apigateway.MethodOptions.html)_.

When  _authorizationType_ is not defined, 'NONE' is the default value.

As specified in the original implementation ticket, _authorizationType_ can be defined either directly on each  _[Method](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-apigateway.Method.htm)_ or via default configurations on the _[RestApi](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apigateway.RestApi.html)_ or _[Resource](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-apigateway.Resource.html)_.
Detection of _authorizationType_ inside _MethodOptions_  is missing when its defined using the _authorizer_ property. Note that, _AuthorizationType_ is optional, thus the rule should raise when its not specified.
 
```
test.addMethod(
  "PATCH",
  new apigateway.HttpIntegration("https://example.org"),
  /* options */ { // TP (wrong message) - AuthorizationType is not omitted but defined to NONE
      authorizer:  {
          authorizerId: "authorizerid",
          authorizationType: apigateway.AuthorizationType.NONE
      }
  }
)

test.addMethod(
  "POST",
  new apigateway.HttpIntegration("https://example.org"),
  /* options */ { // FP - AuthorizationType is being defined here
       authorizer:  {
          authorizerId: "authorizerid",
          authorizationType: apigateway.AuthorizationType.IAM
      }
  }
)

test.addMethod(
  "PUT",
  new apigateway.HttpIntegration("https://example.org"),
  /* options */ {  // TP
      authorizer:  {
          authorizerId: "authorizerid"
      }
  }
)
```
### **How**
Add additional logic to the [checkResourceMethod](https://github.com/SonarSource/SonarJS/pull/3522/files#diff-c4a9b3e9dc2cd4ab293e897afda17866b566179b861a940c4c62434b7fa9123aR84) function for the authorizerid validation.

### **Resources**
https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-apigateway.MethodOptions.html
https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-apigateway.MethodOptions.html#authorizer
https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-apigateway.IAuthorizer.html
